pkg/kubelet/util/util_unix.go

```diff
@@ -135,3 +135,14 @@ func LocalEndpoint(path, file string) (string, error) {
 	}
 	return filepath.Join(u.String(), file+".sock"), nil
 }
+
+// IsUnixDomainSocket returns whether a given file is a AF_UNIX socket file
+func IsUnixDomainSocket(fileInfo os.FileInfo, filePath string) (bool, error) {
+	if fileInfo == nil {
```

Comment:
do we really need this check?

Reply:
Yes - mainly to avoid dereferencing a `nil` FileInfo.

Reply:
for non existent file, shouldn't os.stat() return error?

Reply:
I was thinking that we should check for the case if the caller failed to check the err from os.stat() and passed the nil  fileInfo through. However, given we will do a stat here in this function (as suggested in https://github.com/kubernetes/kubernetes/pull/81397/files#r313746650), this should go away.